roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,153 @@
+        roi_ids = self._dataset_file['estimates']['A']['indices']
+        masks = self._dataset_file['estimates']['A']['data']
+        ids = self._dataset_file['estimates']['A']['indptr']
+        return masks, roi_ids, ids
+
+    def _trace_extractor_read(self, field):
+        extracted_signals = self._dataset_file['estimates'][field] # lazy read dataset)
+        return extracted_signals
+
+    def _summary_image_read(self):
+        if self._dataset_file['estimates'].get('Cn'):
+            summary_images_ = self._dataset_file['estimates']['Cn']
+            return np.array(summary_images_).T
+        else:
+            return None
+
+    def get_accepted_list(self):
+        accepted = self._dataset_file['estimates']['idx_components']
+        if len(accepted.shape)==0:
+            accepted = list(range(self.no_rois))
+        return accepted
+
+    def get_rejected_list(self):
+        return [a for a in range(self.no_rois) if a not in set(self.get_accepted_list())]
+
+    @property
+    def roi_locations(self):
+        _masks, _mask_roi_ids, _mask_ids = self._image_mask_sparse_read()
+        roi_location = np.ndarray([2, self.no_rois], dtype='int')
+        for i in range(self.no_rois):
+            max_mask_roi_id = _mask_roi_ids[_mask_ids[i]+np.argmax(
+                _masks[_mask_ids[i]:_mask_ids[i+1]]
+            )]
+            roi_location[:, i] = [((max_mask_roi_id+1)%(self.image_size[0]+1))-1,#assuming order='F'
+                                  ((max_mask_roi_id+1)//(self.image_size[0]+1))]
+            if roi_location[0,i]<0:
+                roi_location[0,i]=0
+        return roi_location
+
+    @staticmethod
+    def write_segmentation(segmentation_object, savepath):
+        raise NotImplementedError
+
+    # defining the abstract class enformed methods:
+    def get_roi_ids(self):
+        return list(range(self.no_rois))
+
+    def get_num_rois(self):
+        return self._roi_response.shape[0]
+
+    def get_roi_locations(self, roi_ids=None):
+        if roi_ids is None:
+            return self.roi_locations
+        else:
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
+            ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
+            roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
+            return self.roi_locations[:, roi_idx_]
+
+    def get_num_frames(self):
+        return self._roi_response.shape[1]
+
+    def get_traces(self, roi_ids=None, start_frame=None, end_frame=None, name='Fluorescence'):
+        if roi_ids is None:
+            roi_idx_ = range(self.get_num_rois())
+        else:
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
+            ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
+            roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
+        traces = [getattr(self, i) for i in self.__dict__.keys() if name.lower() in i]
```

Comment:
what's going on here? This seems pretty complex. You can just use `traces = getattr(self, name)`?

Reply:
I'm just trying to map the desired trace ( like 'Fluorescence') to the corresponding internal variable that stores this trace
It just frees us to use internal names that could be more programmer friendly than user friendly: "`Fluorescence`" maps to `self._roi_response_fluorescence`, 'Neuropil' to `self._roi_response_neuropil` ..

Reply:
I'm using __dict__ to get a list of all attribute names to search over. Is there a better way of doing this?

Reply:
oh.. so you are searching if `'Fluorescence'.lower()` is in `'_roi_response_fluorescence'`. OK, now I understand. I think this will likely cause problems if there is another class attribute that contains 'fluorescence' in the name for some reason. If you feel there is a need, you can create a dictionary that contains e.g. `{'Fluorescence': '_roi_response_fluorescence'}`